superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -91,16 +89,30 @@ class VisualizeModal extends React.PureComponent {
     });
     if (this.state.chartType === null) {
       hints.push(VISUALIZE_VALIDATION_ERRORS.REQUIRE_CHART_TYPE);
-    } else if (this.state.chartType.requiresTime) {
-      let hasTime = false;
-      for (const colName in cols) {
-        const col = cols[colName];
-        if (col.hasOwnProperty('is_date') && col.is_date) {
-          hasTime = true;
+    } else {
+      if (this.state.chartType.requiresTime) {
+        let hasTime = false;
+        for (const colName in cols) {
+          const col = cols[colName];
+          if (col.hasOwnProperty('is_date') && col.is_date) {
```

Comment:
why do we need both of these conditions? could we just check
`if (col.is_date) {}`